tools/gfx/flag-combiner.h

```diff
@@ -0,0 +1,62 @@
+#ifndef GFX_FLAG_COMBINER_H
+#define GFX_FLAG_COMBINER_H
+
+#include "../../source/core/list.h"
+
+namespace gfx {
+
+/* A default set of flags that can be used for checking devices */
+typedef uint32_t DeviceCheckFlags;
+struct DeviceCheckFlag
+{
+    enum Enum : DeviceCheckFlags
+    {
+        UseFullFeatureLevel = 0x1,      //< If set will use full feature level (on dx this is  D3D_FEATURE_LEVEL_11_1 else will try D3D_FEATURE_LEVEL_11_0)
```

Comment:
The `UseFullFeatureLevel` bit is a little more complex than the name implies, since it is really more like "*request* full feature level." Just because we put `D3D_FEATURE_LEVEL_11_1` in doesn't mean we will get it, and it is possible for the device creation to succeed but give us a lower feature level. It also seems a bit odd that this flag is highly specific to how we do device creation for D3D11, and doesn't really apply to the other cases.